src/main/scala/org/broadinstitute/hail/sparkextras/OrderedPartitioner.scala

```diff
@@ -83,18 +85,18 @@ case class OrderedPartitioner[PK, K](rangeBounds: Array[PK], ascending: Boolean
   }
 
   def mapMonotonic[K2](implicit k2Ok: OrderedKey[PK, K2]): OrderedPartitioner[PK, K2] = {
-    new OrderedPartitioner(rangeBounds, ascending)
+    new OrderedPartitioner(rangeBounds, nPartitions, ascending)
   }
 }
 
 object OrderedPartitioner {
   def empty[PK, K](implicit kOk: OrderedKey[PK, K]): OrderedPartitioner[PK, K] =
-    new OrderedPartitioner[PK, K](Array.empty(kOk.pkct))
+    new OrderedPartitioner[PK, K](Array.empty(kOk.pkct), 0)
 
   def read[PK, K](in: ObjectInputStream)(implicit kOk: OrderedKey[PK, K]): OrderedPartitioner[PK, K] = {
     val ascending = in.readBoolean()
     val rangeBounds = in.readObject().asInstanceOf[Array[PK]]
-    OrderedPartitioner(rangeBounds, ascending)
+    OrderedPartitioner(rangeBounds, rangeBounds.length + 1, ascending)
```

Comment:
Isn't this the same as the original `def nPartitions = rangeBounds.length + 1` bug?  You should read in the RDD in `VSM.read` and then pass in the number of partitions.